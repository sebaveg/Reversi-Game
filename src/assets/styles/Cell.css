.cell {
  border: 1px solid white;
  color: rgba(255, 255, 255, .2);
  cursor: pointer;
  height: 70px;
  width: 70px;
}
.cell:hover {
  color: white;
}
.highlight {
  border: 1px solid yellow;
  color: yellow;
  cursor: pointer;
  height: 70px;
  width: 70px;
  cursor: cell;
}
.highlight:hover {
  border: 1px solid gold;
  color: gold;
}
.black, .white {
  height: 55px;
  width: 55px;
  border: 2px solid grey;
  border-radius: 50%;
  margin: 0 auto;
  cursor: default;
}

.black {
  background-color: black;
}

.white {
  background-color: white;
}

@media (max-width: 1020px) {
  .cell {
    height: 60px;
    width: 60px;
  }

  .highlight {
    height: 60px;
    width: 60px;
  }

  .black,
  .white {
    height: 45px;
    width: 45px;
  }
}

@media (max-width: 720px) {
  .cell {
    height: 50px;
    width: 50px;
  }
  .highlight {
    height: 50px;
    width: 50px;
  }
  .black,
  .white {
    height: 35px;
    width: 35px;
  }
}

@media (max-width: 520px) {
  .cell {
    height: 35px;
    width: 35px;
  }
  .highlight {
    height: 35px;
    width: 35px;
  }
  .black,
  .white {
    height: 25px;
    width: 25px;
  }
}