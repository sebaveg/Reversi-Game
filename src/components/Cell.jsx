/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
import React from 'react';

const Cell = ({
  disk, allowed, children, onClick,
}) => (
  <td className={allowed.length > 0 ? 'highlight' : 'cell'} onClick={onClick}>
    {disk !== 'black' && disk !== 'white' ? children : null}
    <div className={disk} />
  </td>
);

export default Cell;
