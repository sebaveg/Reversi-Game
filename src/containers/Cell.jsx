import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
  setError,
  setBoard,
  changeTurn,
  putDisks,
} from '../actions';

import CellLayout from '../components/Cell';

import '../assets/styles/Cell.css';

class Cell extends Component {
  diskOponent = () => (this.props.currentPlayer === 'white' ? 'black' : 'white')

  inBoard = (x, y) => (x >= 0) && (x < 8) && (y >= 0) && (y < 8)

  // reverse disks oponent
  async reverse() {
    if (this.props.allowed.length > 0) {
      const { board } = this.props;
      const x = this.props.position[0];
      const y = this.props.position[1];
      // await this.props.putDisks({
      //   x, y, disk: this.props.currentPlayer, allowedCells: this.props.board[x][y].allowedCell,
      // });
      // board[x][y].disk = this.props.currentPlayer;
      board[x][y].allowedCell.forEach((cell) => {
        this.props.putDisks({ x: cell.X, y: cell.Y, disk: this.props.currentPlayer });
      });
      // this.props.setBoard(board);
      await this.props.changeTurn();
    } else {
      this.props.setError('No can put your disk here');
      setTimeout(() => {
        this.props.setError('');
      }, 2000);
    }
  }

  render() {
    return (
      <CellLayout
        disk={this.props.disk}
        allowed={this.props.allowed}
        onClick={this.reverse.bind(this)}
      >
        {this.props.children}
      </CellLayout>
    );
  }
}

const mapStateToProps = (state) => ({
  board: state.board.present.board,
  currentPlayer: state.players.present.currentPlayer,
});

const mapDispatchToProps = {
  setError,
  setBoard,
  changeTurn,
  putDisks,
};

export default connect(mapStateToProps, mapDispatchToProps)(Cell);
