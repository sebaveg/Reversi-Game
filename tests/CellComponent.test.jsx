/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable no-undef */
import React from 'react';
import Cell from '../src/components/Cell';

const setUp = (props = {}) => {
  const component = shallow(<Cell {...props} />);
  return component;
};

describe('<Cell /> Component', () => {
  describe('Have props', () => {
    let wrapper;

    beforeEach(() => {
      const props = {
        disk: 'black',
        allowed: [0, 0],
      };
      wrapper = setUp(props);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should render one <td>', () => {
      const component = wrapper.find('td');
      expect(component.length).toBe(1);
    });

    it('should render one <div>', () => {
      const component = wrapper.find('div');
      expect(component.length).toBe(1);
    });

    describe('Have NO props', () => {
      let wrapper;

      beforeEach(() => {
        wrapper = setUp();
      });
    });
  });
});
