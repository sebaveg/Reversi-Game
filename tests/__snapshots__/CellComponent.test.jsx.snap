// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Cell /> Component Have props renders correctly 1`] = `
<td
  className="highlight"
>
  <div
    className="black"
  />
</td>
`;
